02-peer-protocol.md

```diff
@@ -445,6 +446,185 @@ to broadcast the commitment transaction to get his funds back and open a new
 confirms in the next 2016 blocks.
 
+## Channel Quiescence
+
+Various fundamental changes, in particular protocol upgrades, are
+easiest on channels where both commitment transactions match, and no
+pending updates are in flight.  We define a protocol to quiesce the
+channel by indicating that "SomeThing Fundamental is Underway".
+
+### `stfu`
+
+1. type: 2 (`stfu`)
+2. data:
+    * [`channel_id`:`channel_id`]
+    * [`u8`:`initiator`]
+
+### Requirements
+
+The sender of `stfu`:
+  - MUST NOT send `stfu` if any of the sender's htlc additions, htlc removals
+    or fee updates are pending for either peer.
+  - MUST NOT send `stfu` twice.
+  - if it is replying to an `stfu`:
+    - MUST set `initiator` to 0
+  - otherwise:
+    - MUST set `initiator` to 1
+  - MUST set `channel_id` to the id of the channel to quiesce.
+  - MUST now consider the channel to be quiescing.
+  - MUST NOT send an update message after `stfu`.
+
+The receiver of `stfu`:
+  - if it has sent `stfu` then:
+    - MUST now consider the channel to be quiescent
+  - otherwise:
+    - SHOULD NOT send any more update messages.
+    - MUST reply with `stfu` once it can do so.
+
+Upon disconnection:
+  - the channel is no longer considered quiescent.
+
+### Rationale
+
+The normal use would be to cease sending updates, then wait for all
+the current updates to be acknowledged by both peers, then start
+quiescence.  For some protocols, choosing the initiator matters,
+so this flag is sent.
+
+If both sides send `stfu` simultaneously, they will both set
+`initiator` to `1`, in which case the "initiator" is arbitrarily
+considered to be the channel funder (the sender of `open_channel`).
+The quiescence effect is exactly the same as if one had replied to the
+other.
+
+## Splicing
+
+Splicing is the term given for replacing the funding transaction with
+a new one.  For simplicity, splicing takes place once a channel is
+[quiescent](#channel-quiescence).
+
+Operation returns to normal once negotiation is done (while waiting
+for the splice transaction(s) to confirm).
+
+The splice is finally terminated when both sides send
+`splice_complete` to indicate that one of the splice transactions
+reached acceptable depth.
+
+### The `splice` Message
+
+1. type: 74 (`splice`)
+2. data:
+   * [`chain_hash`:`chain_hash`]
+   * [`channel_id`:`channel_id`]
+   * [`u64`:`funding_satoshis`]
+   * [`u32`:`funding_feerate_perkw`]
+   * [`u32`:`locktime`]
+   * [`point`:`funding_pubkey`]
+
+1. type: 76 (`splice_ack`)
+2. data:
+   * [`chain_hash`:`chain_hash`]
+   * [`channel_id`:`channel_id`]
+   * [`u64`:`funding_satoshis`]
+   * [`point`:`funding_pubkey`]
+
+`funding_satoshis` is the amount the sender is putting into the
+post-splice channel. It includes their old funded channel amount
+plus any satoshis they intend to add, or, less any satoshis they
+intend to remove.
+
+#### Requirements
+
+The sender of `splice`:
+- MUST have successfully initiated quiescence.
```

Comment:
to clarify: does that mean that only the initiator of `stfu` will be allowed to advance to send `splice`, or any other protocol for that matter?  (i.e. if Alice sends `stfu` to Bob, only Alice can send `splice`)

what happens in the event where Alice could send `stfu`, refuse to advance to the `splice` (or any other protocol) -- will they be taking the channel hostage?

Reply:
Yes the initiator should be the one to start the splice. I like this approach because keeping turn order explicit keeps the protocol clean and reduces complexity.

Starting `stfu` and hanging is equivalent to hanging on a channel generally. Nodes should go through their normal logic for a hanging node (ie. reconnect after a timeout, eventually force close). `stfu` mode should be reset on reconnection.

Reply:
> what happens in the event where Alice could send stfu, refuse to advance to the splice (or any other protocol) -- will they be taking the channel hostage?

As @ddustin mentions, your channel peer can choose to stop responding at any point (after receiving `add_htlc` for ex) and produce an equivalent result: the channel becomes useless and any in-flight payments are stuck.

`stfu` is no different, and it protocol timing failures (takes too long to advance etc) should be dealt with in the same manner that your channel currently handles other failures to respond.